
From: Markus Lidel <Markus.Lidel@shadowconnect.com>

Okay, appended a patch, which fixes it (at least on my system :-D).

And it also fixes a missing /sys/block/i2o!hdX/device entry.

Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/message/i2o/block-osm.c |    2 ++
 1 files changed, 2 insertions(+)

diff -puN drivers/message/i2o/block-osm.c~i2o-devfs-fix drivers/message/i2o/block-osm.c
--- 25/drivers/message/i2o/block-osm.c~i2o-devfs-fix	Wed Aug  4 15:56:40 2004
+++ 25-akpm/drivers/message/i2o/block-osm.c	Wed Aug  4 15:56:40 2004
@@ -281,6 +281,8 @@ static int i2o_block_probe(struct device
 	gd = i2o_blk_dev->gd;
 	gd->first_minor = unit<<4;
 	sprintf(gd->disk_name, "i2o/hd%c", 'a' + unit);
+	sprintf(gd->devfs_name, "i2o/hd%c", 'a' + unit);
+	gd->driverfs_dev = &i2o_dev->device;
 
 	/* setup request queue */
 	queue = gd->queue;
_
